src/Formats/JSONUtils.cpp

```diff
@@ -286,11 +291,78 @@ namespace JSONUtils
                 return true;
             }
 
-            if (as_nullable)
-                return SerializationNullable::deserializeNullAsDefaultOrNestedTextJSON(column, in, format_settings, serialization);
+            auto do_deserialize = [](IColumn & column_, ReadBuffer & buf_, auto && check_for_empty_string, auto && deserialize) -> bool
+            {
+                if (check_for_empty_string(buf_))
+                {
+                    column_.insertDefault();
+                    return false;
+                }
+                else
+                    return deserialize(column_, buf_);
+            };
+
+            auto deserialize_impl = [as_nullable, &format_settings, &serialization](IColumn & column_, ReadBuffer & buf_) -> bool
+            {
+                if (as_nullable)
+                    return SerializationNullable::deserializeNullAsDefaultOrNestedTextJSON(column_, buf_, format_settings, serialization);
+
+                serialization->deserializeTextJSON(column_, buf_, format_settings);
+                return true;
+            };
+
+            if (!format_settings.json.empty_as_default  || in.eof() || *in.position() != EMPTY_STRING[0])
+                return do_deserialize(column, in, [](ReadBuffer &) { return false; }, deserialize_impl);
+
+            if (in.available() >= EMPTY_STRING_LENGTH)
+            {
+                /// We have enough data in buffer to check if we have an empty string.
+                auto check_for_empty_string = [](ReadBuffer & buf_)
+                {
+                    auto * pos = buf_.position();
+                    if (checkString(EMPTY_STRING, buf_))
+                        return true;
+                    else
+                    {
+                        buf_.position() = pos;
+                        return false;
+                    }
```

Comment:
```suggestion
                    buf_.position() = pos;
                    return false;
```

Reply:
Done. I did the same in a couple of other places since I noticed this is the accepted style here.